src/grid/cell.component.ts

```diff
@@ -535,20 +535,16 @@ export class IgxGridCellComponent implements IGridBus, OnInit, OnDestroy {
         }
 
         if (target) {
-            const targetRowOffset = target.row.nativeElement.offsetTop;
-            const containerOffset = target.grid.verticalScrollContainer.dc.instance._viewContainer.element.nativeElement.offsetTop;
-
-            if (targetRowOffset === 0 && containerOffset < 0) {
-                // Target is part of the first row in the container that is partially visible
-                verticalScroll.scrollTop += containerOffset;
-                const oldChunkIndex = target.row.grid.verticalScrollContainer.state.startIndex;
-
+            const containerTopOffset =
+                parseInt(this.row.grid.verticalScrollContainer.dc.instance._viewContainer.element.nativeElement.style.top, 10);
+            if (this.grid.rowHeight > -containerTopOffset // not the entire row is visible, due to grid offset
+                && verticalScroll.scrollTop // the scrollbar is not at the first item
+                && target.row.element.nativeElement.offsetTop < this.grid.rowHeight) { // the target is in the first row
+                verticalScroll.scrollTop -= this.grid.rowHeight;
                 this.row.grid.verticalScrollContainer.onChunkLoad.pipe(take(1)).subscribe({
                     next: (e: any) => {
-                        if (oldChunkIndex === e.startIndex) {
-                            target.nativeElement.focus();
-                        }
-                        this.row.cdr.detectChanges();
+                        const cell = this.gridAPI.get_cell_by_visible_index(this.gridID, rowIndex, this.visibleColumnIndex);
```

Comment:
What @kdinev is requesting, is that we remove duplication of those chunkload subscribe functions. We need a centralized function for that. Any other logic, inside keydown/up handlers that is used on more than one place, should be refactored. Thanks!